splash-reward-distributor/src/engine/mod.rs

```diff
@@ -263,14 +275,18 @@ where
     }
     match executor.execute().await {
         Ok(res) => {
-            // TODO: index this with executed tasks (DEX-919)
-            let _tx_hash = res.output;
+            let tx_hash = res.output;
+
+            task_ids_by_tx_hash
+                .insert(tx_hash, res.executed_tasks.clone())
+                .await;
```

Comment:
The problem with this solution is that index write and update of tasks' statuses in the queue are not atomic. Think of what will happen if the bot halts after index write happened but queue didn't received updates yet. I suggest to maintain this index inside TaskQueue so everything is written atomically, task->tx_id relations may be passed into the queue via the `QueueCmd::Done(task, tx_id)` command.

Reply:
Done.